src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,39 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userByEmail = userService.findByEmail(email);
+        if (userByEmail.isEmpty() || !userByEmail.get().getPassword()
+                .equals(HashUtil.hashPassword(password, userByEmail.get().getSalt()))) {
+            throw new AuthenticationException("There is no user with this email or password.");
```

Comment:
```suggestion
            throw new AuthenticationException("There is no such user with email: " + email);
```

Reply:
@Nick97-git i don't think, that it will be secure

Reply:
if you'd add password yes, but without it, it's ok